### PR TITLE
Remove ConsensusParams.TxSize and ConsensusParams.BlockGossip

### PR DESCRIPTION
**Pull request by melekes**
Refs #2347

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: abci/types/types.proto [hunk 1]
```diff
@@ -200,27 +200,21 @@ message ResponseCommit {
...
 message ConsensusParams {
   BlockSize block_size = 1;
-  TxSize tx_size = 2;
-  BlockGossip block_gossip = 3;
+  EvidenceParams evidence_params = 2;
 }
 
 // BlockSize contains limits on the block size.
 message BlockSize {
+  // Note: must be greater than 0
```
Comment (ValarDragon) [inline abci/types/types.proto]: Couldn't it be -1?

File: blockchain/reactor_test.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func newBlockchainReactor(logger log.Logger, maxBlockHeight int64) *BlockchainRe
 	for blockHeight := int64(1); blockHeight <= maxBlockHeight; blockHeight++ {
 		firstBlock := makeBlock(blockHeight, state)
 		secondBlock := makeBlock(blockHeight+1, state)
-		firstParts := firstBlock.MakePartSet(state.ConsensusParams.BlockGossip.BlockPartSizeBytes)
+		firstParts := firstBlock.MakePartSet(types.BlockPartSizeBytes)
```
Comment (ValarDragon) [inline blockchain/reactor_test.go]: I still don't really understand why this isn't a consensus parameter. What optimizations are we getting by optimizing around 1MB blocks, it feels like we could still leave this as parameterized to me. (Or perhaps parameterize as "number of block parts")

Comment (melekes) [reply blockchain/reactor_test.go]: NOTE: `BlockPartSizeBytes` really depends on the network bandwidth and network topology. It should be possible to make it adaptive to the current env by analysing the network, but it's #post-launch imho.

Plus we wasn't checking the size when accepting block part (no consensus per se 😄 ).

File: abci/types/types.proto (hunk 1 above)
Comment (melekes) [reply abci/types/types.proto]: Unlimited? Do you think it's a good idea?

File: types/params.go [hunk 3]
```diff
@@ -46,70 +36,56 @@ type EvidenceParams struct {
...
+			params.BlockSize.MaxBytes)
 	}
-
-	// ensure blocks aren't too big
 	if params.BlockSize.MaxBytes > MaxBlockSizeBytes {
 		return cmn.NewError("BlockSize.MaxBytes is too big. %d > %d",
 			params.BlockSize.MaxBytes, MaxBlockSizeBytes)
 	}
+
+	if params.BlockSize.MaxGas < -1 {
```
Comment (ebuchman) [inline types/params.go]: Maybe these should all be the same for consistency? ie. `< -1`

File: types/params.go [hunk 4]
```diff
@@ -126,25 +102,11 @@ func (params ConsensusParams) Update(params2 *abci.ConsensusParams) ConsensusPar
...
-		}
-	}
-	if params2.TxSize != nil {
-		if params2.TxSize.MaxBytes > 0 {
-			res.TxSize.MaxBytes = int(params2.TxSize.MaxBytes)
-		}
-		if params2.TxSize.MaxGas > 0 {
-			res.TxSize.MaxGas = params2.TxSize.MaxGas
-		}
+		res.BlockSize.MaxBytes = int(params2.BlockSize.MaxBytes)
```
Comment (ebuchman) [inline types/params.go]: can we unify these and remove the cast?

File: types/params_test.go [hunk 5]
```diff
@@ -9,10 +9,10 @@ import (
 	abci "github.com/tendermint/tendermint/abci/types"
 )
 
-func newConsensusParams(txsBytes, partSize int) ConsensusParams {
+func newConsensusParams(txsBytes, evidenceAge int) ConsensusParams {
```
Comment (ebuchman) [inline types/params_test.go]: - txBytes should be called maxBytes or blockBytes
- we should also pass in maxGas or blockGas

File: types/params_test.go [hunk 6]
```diff
@@ -21,50 +21,45 @@ func TestConsensusParamsValidation(t *testing.T) {
...
+			assert.NoErrorf(t, tc.params.Validate(), "expected no error for valid params (#%d)", i)
 		} else {
-			assert.Error(t, tc.params.Validate(), "expected error for non valid params")
+			assert.Errorf(t, tc.params.Validate(), "expected error for non valid params (#%d)", i)
 		}
 	}
 }
 
-func makeParams(txsBytes, blockGas, txBytes, txGas, partSize int) ConsensusParams {
+func makeParams(txsBytes, blockGas, evidenceAge int) ConsensusParams {
```
Comment (ebuchman) [inline types/params_test.go]: deduplicate with newConsensusParams?

File: abci/types/types.proto (hunk 1 above)
Comment (ebuchman) [reply abci/types/types.proto]: Probably best if we're consistent, so these ints all have the same domain

**Comment by ebuchman** [comment]
Follow up in https://github.com/tendermint/tendermint/issues/2382

**Pull request merged**